substrate/frame/nomination-pools/src/lib.rs

```diff
@@ -2300,7 +2302,7 @@ pub mod pallet {
 
 			let slash_weight =
 				// apply slash if any before withdraw.
-				match Self::do_apply_slash(&member_account, None) {
+				match Self::do_apply_slash(&member_account, None, false) {
```

Comment:
My previous comment only made sense when taken with another made to this line, which asked: why don't we want to enforce the ED checks here?

Sorry, I had the GH open for a while, it might have been lost when I refreshed the browser.

Reply:
At this point, a gas fee is charged for slashing. Additionally, we need to clean up all bookkeeping before a member withdraws.